controllers/iochaos/fs/types.go

```diff
@@ -161,29 +162,31 @@ func (r *Reconciler) cleanFinalizersAndRecover(ctx context.Context, iochaos *v1a
 func (r *Reconciler) recoverPod(ctx context.Context, pod *v1.Pod, iochaos *v1alpha1.IoChaos) error {
 	r.Log.Info("Recovering", "namespace", pod.Namespace, "name", pod.Name)
 
-	var ns v1.Namespace
-	if err := r.Get(ctx, types.NamespacedName{Name: pod.Namespace}, &ns); err != nil {
-		return err
-	}
+	cctx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+	defer cancel()
+	err := wait.PollUntil(2*time.Second, func() (bool, error) {
+		if err := r.recoverInjectAction(ctx, pod, iochaos); err != nil {
+			if utils.IsCaredNetError(err) {
+				r.Log.Info("Recover I/O chaos action, network is not ok, retrying...",
+					"namespace", pod.Namespace, "name", pod.Name)
+				return false, nil
+			}
 
-	annotations := ns.GetAnnotations()
-	if annotations == nil {
-		annotations = make(map[string]string)
-	}
+			return false, err
+		}
 
-	if _, ok := annotations[v1alpha1.WebhookInitPodAnnotationKey]; ok {
-		return r.recoverInjectAction(ctx, pod, iochaos)
-	}
+		r.Log.Info("Recover I/O chaos action successfully")
 
-	if err := utils.UnsetIoInjection(ctx, r.Client, pod, iochaos); err != nil {
-		r.Log.Error(err, "failed to unset I/O injection",
+		return true, nil
+	}, cctx.Done())
```

Comment:
When network is not ok, Poll loop and `cctx` timeout, `wait.PollUntil` will return timeout err or nil?

Reply:
timeout err